### PR TITLE
Add classes for OOP school library

### PR DESCRIPTION
**Pull request by why-a-en**
# Branch changes

- Added Person class
- Added Student class
- Added Teacher class

### CODE REVIEW

File: classes/student.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(classroom)
+    super(name, age, parent_permission)
```
Comment (julie-ify) [inline classes/student.rb]: - [x] Kindly pass in the other parameters for creating a person. We need to pass in the name, age and parent_permission.

- [x] We also need to follow the order of the parameters in the `#super` method.

  ```
  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
  ```

File: classes/teacher.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(specialization)
+    super(name, age)
```
Comment (julie-ify) [inline classes/teacher.rb]: - [x] Kindly pass in the other parameters for creating a person. We need to pass in the name and age.

- [x] We also need to follow the order of the parameters in the `#super` method.

  ```
  def initialize(specialization, age, name = 'Unknown')
    super(age, name)
  ```

File: classes/person.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_accessor :@name, :@age
+  attr_reader :@id
```
Comment (julie-ify) [inline classes/person.rb]: - [x] Defining your getter and setter this way returns errors. This is what you should have:

  ```
  attr_accessor :name, :age
  attr_reader :id
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @anbehindY,

Good job so far!

## Highlights!
- Good code organization and class inheritance✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/person.rb (hunk 3 above)
Comment (why-a-en) [reply classes/person.rb]: thank you so much @julie-ify

**Review by amejid — APPROVED** [review]
Hi @anbehindY, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Linter check passing :heavy_check_mark: 
- Correct workflow :heavy_check_mark: 
- Inheritance successfully used :heavy_check_mark: 
- Changes implemented :heavy_check_mark: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**